### ISSUE TITLE
Is it normal for the train loss to often be negative?

### ISSUE THREAD

**Issue by supersuperdong** [marco-rudolph/differnet#45]
Furthermore, it seems that the train loss here is only seeking the parameters of the normalizing flow (NF) structure that can most likely transform the features y into a standard normal distribution z. How do we determine whether the feature extraction network is good or not?

**Comment by marco-rudolph** [comment]
Regarding the loss: https://github.com/marco-rudolph/differnet/issues/40
Regarding the features: You cannot really determine if the used features are helpful without having any anomalies - or creating synthetical ones. We use some kind of universal feature extractor by taking an ImageNet-pretrained network, which has shown to be effective in our experiments.
